corehq/apps/reports/standard/deployments.py

```diff
@@ -318,22 +324,33 @@ def process_rows(self, users, fmt_for_export=False):
                 last_builds = reporting_metadata.get('last_builds')
                 if last_builds:
                     last_build = self.get_data_for_app(last_builds, self.selected_app_id)
+                if device:
+                    device_app_metas = device.get('app_meta')
+                    if device_app_metas:
+                        device_app_meta = self.get_data_for_app(device_app_metas, self.selected_app_id)
             else:
                 last_sub = reporting_metadata.get('last_submission_for_user', {})
                 last_sync = reporting_metadata.get('last_sync_for_user', {})
                 last_build = reporting_metadata.get('last_build_for_user', {})
+                app_id = None
+                if last_build:
+                    app_id = last_build.get('app_id')
+                if device is not None and app_id is not None:
+                    device_app_metas = device.get('app_meta')
+                    if device_app_metas:
+                        device_app_meta = self.get_data_for_app(device_app_metas, app_id)
+
             if last_sub and last_sub.get('commcare_version'):
                 commcare_version = _get_commcare_version(last_sub.get('commcare_version'))
             else:
-                devices = user.get('devices', None)
-                if devices:
-                    device = max(devices, key=lambda dev: dev['last_used'])
-                    if device.get('commcare_version', None):
-                        commcare_version = _get_commcare_version(device['commcare_version'])
+                if device and device.get('commcare_version', None):
+                    commcare_version = _get_commcare_version(device['commcare_version'])
             if last_sub and last_sub.get('submission_date'):
                 last_seen = string_to_utc_datetime(last_sub['submission_date'])
             if last_sync and last_sync.get('sync_date'):
                 last_sync_date = string_to_utc_datetime(last_sync['sync_date'])
+            if device_app_meta and device_app_meta.get('num_unsent_forms') is not None:
+                num_unsent_forms = str(device_app_meta['num_unsent_forms'])
```

Comment:
Why is this count cast to a string?

Reply:
this was done to display 0. There is a condition below like `num_unsent_forms or '---'` which would always go to --- if num_unsent_forms was int. Changed it to a if condition and removed casting.